environment-api/src/main/java/com/sequenceiq/environment/api/v1/platformresource/EnvironmentPlatformResourceEndpoint.java

```diff
@@ -0,0 +1,163 @@
+package com.sequenceiq.environment.api.v1.platformresource;
+
+import static com.sequenceiq.environment.api.doc.ModelDescriptions.CONNECTOR_NOTES;
+
+import javax.validation.constraints.NotEmpty;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+
+import com.sequenceiq.cloudbreak.jerseyclient.RetryAndMetrics;
+import com.sequenceiq.common.api.type.CdpResourceType;
+import com.sequenceiq.environment.api.v1.platformresource.PlatformResourceModelDescription.OpEnvDescription;
+import com.sequenceiq.environment.api.v1.platformresource.model.AccessConfigTypeQueryParam;
+import com.sequenceiq.environment.api.v1.platformresource.model.PlatformAccessConfigsResponse;
+import com.sequenceiq.environment.api.v1.platformresource.model.PlatformEncryptionKeysResponse;
+import com.sequenceiq.environment.api.v1.platformresource.model.PlatformGatewaysResponse;
+import com.sequenceiq.environment.api.v1.platformresource.model.PlatformIpPoolsResponse;
+import com.sequenceiq.environment.api.v1.platformresource.model.PlatformNetworksResponse;
+import com.sequenceiq.environment.api.v1.platformresource.model.PlatformNoSqlTablesResponse;
+import com.sequenceiq.environment.api.v1.platformresource.model.PlatformResourceGroupsResponse;
+import com.sequenceiq.environment.api.v1.platformresource.model.PlatformSecurityGroupsResponse;
+import com.sequenceiq.environment.api.v1.platformresource.model.PlatformSshKeysResponse;
+import com.sequenceiq.environment.api.v1.platformresource.model.PlatformVmtypesResponse;
+import com.sequenceiq.environment.api.v1.platformresource.model.RegionResponse;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+
+@RetryAndMetrics
+@Path("/v1/env/platform_resources")
+@Consumes(MediaType.APPLICATION_JSON)
+@Api(value = "/v1/env/platform_resources", description = PlatformResourceModelDescription.CONNECTOR_V1_DESCRIPTION, protocols = "http,https",
+        consumes = MediaType.APPLICATION_JSON)
+public interface EnvironmentPlatformResourceEndpoint {
+
+    @GET
+    @Path("machine_types")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = OpEnvDescription.GET_VMTYPES_BY_CREDENTIAL, produces = MediaType.APPLICATION_JSON, notes = CONNECTOR_NOTES,
+            nickname = "getVmTypesByCredentialByEnv")
+    PlatformVmtypesResponse getVmTypesByCredential(
+            @QueryParam("environmentCrn") @NotEmpty String environmentCrn,
+            @QueryParam("region") String region,
+            @QueryParam("platformVariant") String platformVariant,
+            @QueryParam("availabilityZone") String availabilityZone,
```

Comment:
Aren't these (region, platformVariant, availabilityZone) redundant from environment perspective?

Reply:
Possibly, to be honest, I didn't checked, only duplicated the API with environment CRN change, we can spend more time to check it and make it easier to use